csh_ldap/__init__.py

```diff
@@ -1,11 +1,42 @@
+from functools import wraps
 import ldap
 from ldap.ldapobject import ReconnectLDAPObject
-
 import srvlookup
-
 from csh_ldap.member import CSHMember
 from csh_ldap.group import CSHGroup
 
+MAX_RECONNECTS = 3
+
+
+def reconnect_on_fail(method):
+    @wraps(method)
+    def wrapper(*method_args, **method_kwargs):
+        max_reconnects = MAX_RECONNECTS
+        self = method_args[0]
+        while max_reconnects:
+            try:
+                result = method(*method_args, **method_kwargs)
+                return result
+            except:
+                ldap_srvs = srvlookup.lookup("ldap", "tcp", self.__domain__)
+                self.ldap_uris = ''.join([f'ldaps://{uri.hostname},' for uri in ldap_srvs])
+                self.ldap_uris = self.ldap_uris.split(',')[2:][:-1]
+                for uri in self.ldap_uris:
```

Comment:
Same question as below about generating, joining, and resplitting a list. Also, what's the purpose of the `[2:][:-1]`?
```suggestion
                self.ldap_uris = [f'ldaps://{uri.hostname},' for uri in ldap_srvs]
                for uri in self.ldap_uris:
```